src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+    private Node<T> newNode;
+    private Node<T> temporaryNode;
+
+    private static class Node<T> {
+        protected T element;
+        protected Node<T> prev;
+        protected Node<T> next;
+
+        protected Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.element = element;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            first = new Node<>(null, null, null);
+            last = new Node<>(null, null, null);
+            temporaryNode = new Node<>(first, value, last);
+            first = temporaryNode;
+            last = temporaryNode;
+            size++;
+            return;
+        }
+        newNode = new Node<>(last, value, null);
+        last.next = newNode;
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndexIfInBound(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return indexToNode(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        temporaryNode = indexToNode(index);
+        T oldRecord = temporaryNode.element;
+        temporaryNode.element = value;
+        return oldRecord;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        temporaryNode = indexToNode(index);
+        unLink(temporaryNode, index);
+        return temporaryNode.element;
     }
 
     @Override
     public boolean remove(T object) {
+        temporaryNode = first;
+        for (int i = 0; i < size; i++) {
+            if (temporaryNode.element == object || (temporaryNode.element != null
+                    && temporaryNode.element.equals(object))) {
+                remove(i);
+                return true;
+            }
+            temporaryNode = temporaryNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndexIfInBound(T value, int index) {
```

Comment:
method named `checkIndexIfInBound` must only check if index is in bounds, no more